### PR TITLE
Storage policy configuration parsing (#65)

### LINKED ISSUE #65: [SP] Configuration file parsing
Implement a utility module which exports a high-level data structure representing the configuration as described in the specification document. This include parsing and writing instances of this structure from/to streams.

* [x] Definition of structure
* [x] Implementation of accessors
* [x] Implementation of parsing and writing
* [x] Tests

### PR DESCRIPTION
**Pull request by NicolasT**
Closes #65

### CODE REVIEW

**Comment by NicolasT** [comment]
Currently WiP (some tasks in #65 left).

- [ ] Configuration unparsing
- [ ] Testing configuration failures (bring coverage to 100%)

**Comment by scality** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://37.187.159.67:5443/job/swift-tox/119/<h2>Build result: FAILURE</span></h2>[...truncated 18 lines...]Triggering swift-tox » py27-swifthead,ubuntu14-largeTriggering swift-tox » py27-swift1.13.1,ubuntu14-largeTriggering swift-tox » py27-swift2.2.0,ubuntu12-largeTriggering swift-tox » py27-swifthead,centos7-largeTriggering swift-tox » py27-swift2.2.0,centos7-largeTriggering swift-tox » py27-swift1.13.1,ubuntu12-largeTriggering swift-tox » py26-swift1.13.1,centos6-largeTriggering swift-tox » pep8,ubuntu14-largeConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: All nodes of label ‘ubuntu12-large’ are offlineConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: ubuntu12-large-c4f is offlineConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: Waiting for next available executor on ubuntu12-large-c4fConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: Waiting for next available executor on ubuntu12-largeswift-tox » py27-swifthead,ubuntu12-large completed with result FAILUREswift-tox » py27-swift2.2.0,ubuntu14-large completed with result SUCCESSswift-tox » py27-swift1.13.1,centos7-large completed with result FAILUREswift-tox » py27-swifthead,ubuntu14-large completed with result FAILUREswift-tox » py27-swift1.13.1,ubuntu14-large completed with result SUCCESSswift-tox » py27-swift2.2.0,ubuntu12-large completed with result FAILUREswift-tox » py27-swifthead,centos7-large completed with result SUCCESSswift-tox » py27-swift2.2.0,centos7-large completed with result FAILUREswift-tox » py27-swift1.13.1,ubuntu12-large completed with result SUCCESSswift-tox » py26-swift1.13.1,centos6-large completed with result FAILUREswift-tox » pep8,ubuntu14-large completed with result SUCCESSSetting status of 8702d85314578dde4f11c84f3f333d9bf0d7c4b7 to FAILURE with url https://37.187.159.67:5443/job/swift-tox/119/ and message: Merged build finished.
Test FAILed.

**Comment by scality** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://37.187.159.67:5443/job/swift-tox/120/<h2>Build result: FAILURE</span></h2>[...truncated 17 lines...]Triggering swift-tox » py27-swift1.13.1,centos7-largeTriggering swift-tox » py27-swifthead,ubuntu14-largeTriggering swift-tox » py27-swift1.13.1,ubuntu14-largeTriggering swift-tox » py27-swift2.2.0,ubuntu12-largeTriggering swift-tox » py27-swifthead,centos7-largeTriggering swift-tox » py27-swift2.2.0,centos7-largeTriggering swift-tox » py27-swift1.13.1,ubuntu12-largeTriggering swift-tox » py26-swift1.13.1,centos6-largeTriggering swift-tox » pep8,ubuntu14-largeConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: All nodes of label ‘ubuntu12-large’ are offlineConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: ubuntu12-large-7b6 is offlineConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: All nodes of label ‘ubuntu12-large’ are offlineswift-tox » py27-swifthead,ubuntu12-large completed with result SUCCESSswift-tox » py27-swift2.2.0,ubuntu14-large completed with result FAILUREswift-tox » py27-swift1.13.1,centos7-large completed with result FAILUREswift-tox » py27-swifthead,ubuntu14-large completed with result SUCCESSswift-tox » py27-swift1.13.1,ubuntu14-large completed with result SUCCESSswift-tox » py27-swift2.2.0,ubuntu12-large completed with result SUCCESSswift-tox » py27-swifthead,centos7-large completed with result SUCCESSswift-tox » py27-swift2.2.0,centos7-large completed with result SUCCESSswift-tox » py27-swift1.13.1,ubuntu12-large completed with result FAILUREswift-tox » py26-swift1.13.1,centos6-large completed with result SUCCESSswift-tox » pep8,ubuntu14-large completed with result SUCCESSSetting status of d63b762dbd61d881e01b6b0b65ea24e7942742a2 to FAILURE with url https://37.187.159.67:5443/job/swift-tox/120/ and message: Merged build finished.
Test FAILed.

**Comment by NicolasT** [comment]
No longer WiP :smiley:

**Comment by scality** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://37.187.159.67:5443/job/swift-tox/121/<h2>Build result: FAILURE</span></h2>[...truncated 17 lines...]Triggering swift-tox » py27-swift1.13.1,centos7-largeTriggering swift-tox » py27-swifthead,ubuntu14-largeTriggering swift-tox » py27-swift1.13.1,ubuntu14-largeTriggering swift-tox » py27-swift2.2.0,ubuntu12-largeTriggering swift-tox » py27-swifthead,centos7-largeTriggering swift-tox » py27-swift2.2.0,centos7-largeTriggering swift-tox » py27-swift1.13.1,ubuntu12-largeTriggering swift-tox » py26-swift1.13.1,centos6-largeTriggering swift-tox » pep8,ubuntu14-largeConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: All nodes of label ‘ubuntu12-large’ are offlineConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: ubuntu12-large-d69 is offlineConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: Waiting for next available executor on ubuntu12-large-d69swift-tox » py27-swifthead,ubuntu12-large completed with result SUCCESSswift-tox » py27-swift2.2.0,ubuntu14-large completed with result SUCCESSswift-tox » py27-swift1.13.1,centos7-large completed with result FAILUREswift-tox » py27-swifthead,ubuntu14-large completed with result FAILUREswift-tox » py27-swift1.13.1,ubuntu14-large completed with result FAILUREswift-tox » py27-swift2.2.0,ubuntu12-large completed with result FAILUREswift-tox » py27-swifthead,centos7-large completed with result SUCCESSswift-tox » py27-swift2.2.0,centos7-large completed with result FAILUREswift-tox » py27-swift1.13.1,ubuntu12-large completed with result FAILUREswift-tox » py26-swift1.13.1,centos6-large completed with result FAILUREswift-tox » pep8,ubuntu14-large completed with result SUCCESSSetting status of 749d7fbc01c688b3a18637f59b1964ab9811a220 to FAILURE with url https://37.187.159.67:5443/job/swift-tox/121/ and message: Merged build finished.
Test FAILed.

File: swift_scality_backend/policy_configuration.py [hunk 1]
```diff
@@ -0,0 +1,612 @@
...
+
+    def __eq__(self, other):
+        if isinstance(other, basestring):
+            return self.url == urlparse.urlparse(other)
+        elif isinstance(other, urlparse.ParseResult):
+            return self.url == other
+        elif isinstance(other, Endpoint):
+            return self.url == other.url
+        else:
+            return NotImplemented
```
Comment (JordanP) [inline swift_scality_backend/policy_configuration.py]: Why return and not raise ?

File: swift_scality_backend/policy_configuration.py [hunk 2]
```diff
@@ -0,0 +1,612 @@
...
+            if not parser.has_option(section, setting):
+                if default is not _default:
+                    return default
+                else:
+                    raise ConfigurationError(
+                        'Section %r lacks a %r setting' % (section, setting))
+
+            return parser.get(section, setting)
+
+        def split_list(val):
```
Comment (JordanP) [inline swift_scality_backend/policy_configuration.py]: I need exactly this to parse the configuration of the swift-object-server, the sproxyd_host flag.

Could you move this function in utils.py

**Comment by JordanP** [comment]
LGTM

**Comment by scality** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://37.187.159.67:5443/job/swift-tox/128/<h2>Build result: FAILURE</span></h2>[...truncated 17 lines...]Triggering swift-tox » py27-swift1.13.1,centos7-largeTriggering swift-tox » py27-swifthead,ubuntu14-largeTriggering swift-tox » py27-swift1.13.1,ubuntu14-largeTriggering swift-tox » py27-swift2.2.0,ubuntu12-largeTriggering swift-tox » py27-swifthead,centos7-largeTriggering swift-tox » py27-swift2.2.0,centos7-largeTriggering swift-tox » py27-swift1.13.1,ubuntu12-largeTriggering swift-tox » py26-swift1.13.1,centos6-largeTriggering swift-tox » pep8,ubuntu14-largeConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: All nodes of label ‘ubuntu12-large’ are offlineConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: ubuntu12-large-331 is offlineConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: Waiting for next available executor on ubuntu12-large-331swift-tox » py27-swifthead,ubuntu12-large completed with result FAILUREswift-tox » py27-swift2.2.0,ubuntu14-large completed with result FAILUREswift-tox » py27-swift1.13.1,centos7-large completed with result FAILUREswift-tox » py27-swifthead,ubuntu14-large completed with result FAILUREswift-tox » py27-swift1.13.1,ubuntu14-large completed with result FAILUREswift-tox » py27-swift2.2.0,ubuntu12-large completed with result FAILUREswift-tox » py27-swifthead,centos7-large completed with result FAILUREswift-tox » py27-swift2.2.0,centos7-large completed with result FAILUREswift-tox » py27-swift1.13.1,ubuntu12-large completed with result FAILUREswift-tox » py26-swift1.13.1,centos6-large completed with result FAILUREswift-tox » pep8,ubuntu14-large completed with result FAILURESetting status of a0bb001f51c6a724ceda44d94d80ee7cbee866d7 to FAILURE with url https://37.187.159.67:5443/job/swift-tox/128/ and message: Merged build finished.
Test FAILed.

**Pull request merged**